scala-package/core/src/test/scala/org/apache/mxnet/IOSuite.scala

```diff
@@ -303,5 +303,32 @@ class IOSuite extends FunSuite with BeforeAndAfterAll {
     assert(dataDesc(0).layout == Layout.NTC)
     assert(labelDesc(0).dtype == DType.Int32)
     assert(labelDesc(0).layout == Layout.NT)
+
+
+    // Test with passing Float64 hardcoded as Dtype of data
+    val dataIter4 = new NDArrayIter(
+      IO.initDataDesc(data, false, "data", DType.Float64, Layout.NTC),
+      IO.initDataDesc(label, false, "label", DType.Int32, Layout.NT),
```

Comment:
Any reason for the label to not also have DType.Float64? I know people don't really use that for labels but we added the capability so should test it as well.

Reply:
The default DType in DataDesc is Float32. In this test, I'm trying to show here that not just Float64, but any other DType would also work with Data Descriptors :) 

And yes, for labels, people don't require higher precision datatypes.